streamx-plugin/streamx-flink-submit/streamx-flink-submit-core/src/main/scala/com/streamxhub/streamx/flink/submit/trait/FlinkSubmitTrait.scala

```diff
@@ -388,18 +386,21 @@ trait FlinkSubmitTrait extends Logger {
 
   /**
    * get dir for default config or params
+   *
    * @param stopRequest requset params
    * @return
    */
-  def getSavePointDir(stopRequest: StopRequest): String = {
-    if (stopRequest.customSavePointPath.isEmpty) {
-      getOptionFromDefaultFlinkConfig[String](
-        stopRequest.flinkVersion.flinkHome,
-        ConfigOptions.key(CheckpointingOptions.SAVEPOINT_DIRECTORY.key())
-          .stringType()
-          .defaultValue(s"${workspace.APP_SAVEPOINTS}")
-      )
-    } else stopRequest.customSavePointPath
+  def getSavePointDir(stopRequest: StopRequest): String = Try(Option(stopRequest.customSavePointPath).get).getOrElse {
+    getOptionFromDefaultFlinkConfig[String](
+      stopRequest.flinkVersion.flinkHome,
+      ConfigOptions.key(CheckpointingOptions.SAVEPOINT_DIRECTORY.key())
+        .stringType()
+        .defaultValue {
+          if (stopRequest.executionMode == ExecutionMode.YARN_APPLICATION) {
```

Comment:
This logic is wrong，You need to set the parameter `stoprequest#withsavepoint` to check

Reply:
> This logic is wrong，You need to set the parameter `stoprequest#withsavepoint` to check

thx, fixed.